src/recipe/test/e2e/createRecipe.controller.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { RecipeService } from '../../recipe.service';
+import { AuthModule } from '../../../auth/auth.module';
+import { PrismaService } from '../../../prisma/prisma.service';
+import { JwtService } from '@nestjs/jwt';
+import { RecipeServiceMock } from '../../../recipe/test/mocks/recipe.service.mock';
+
+describe('Recipe Controller - Create', () => {
+  let app: INestApplication;
+  let prisma: PrismaService;
+  let jwtService: JwtService;
+  let accessToken: string;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [RecipeModule, AuthModule],
+    })
+      .overrideProvider(RecipeService)
+      .useClass(RecipeServiceMock)
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe({ transform: true }));
+    await app.init();
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+    jwtService = moduleFixture.get<JwtService>(JwtService);
+  });
+
+  beforeEach(async () => {
+    const testUser = await prisma.user.findFirst({
+      where: {
+        name: process.env.TEST_NAME,
+      },
+    });
+    accessToken = jwtService.sign({
+      name: testUser.name,
+      sub: testUser.id,
+    });
+  });
+
+  it('should create a recipe', async () => {
+    const payload = {
+      name: 'testName',
+      description: 'testDescription',
+      imageURL: 'testImageURL',
+      preparing: [
+        {
+          step: 'testStep',
+          order: 1,
+        },
+      ],
+      ingredients: [
+        {
+          name: 'testName',
+          amount: 'testAmount',
+        },
+      ],
+    };
+    return request(app.getHttpServer())
+      .post('/recipe')
+      .send(payload)
+      .set('Authorization', `bearer ${accessToken}`)
+      .expect(HttpStatus.CREATED)
+      .then((res) => {
+        console.log(res.body);
+        expect(res.body.id).toBe(1);
+        expect(res.body.name).toBe('testName');
+        expect(res.body.description).toBe('testDescription');
+        expect(res.body.imageURL).toBe('testImageURL');
+        expect(res.body.preparing[0].step).toBe('testStep');
+        expect(res.body.preparing[0].order).toBe(1);
+        expect(res.body.ingredients[0].name).toBe('testName');
+        expect(res.body.ingredients[0].amount).toBe('testAmount');
+      });
+  });
+
+  it('should throw error on unathourized user', async () => {
+    const payload = {
+      name: 'testName',
+      description: 'testDescription',
+      imageURL: 'testImageURL',
+      preparing: [
+        {
+          step: 'testStep',
+          order: 1,
+        },
+      ],
+      ingredients: [
+        {
+          name: 'testName',
+          amount: 'testAmount',
+        },
+      ],
+    };
+    return request(app.getHttpServer())
+      .post('/recipe')
+      .send(payload)
+      .expect(HttpStatus.UNAUTHORIZED);
+  });
+
+  it('should throw validation error when property name is number', async () => {
```

Comment:
You have repeated a lot of code. Try to keep the DRY rule.
take a look at `payload` object. In most cases it uses the same things. Try to move it to some common variable like `validPayload` and use it in other places while changing only the property that you are going to test.

Besides that, the test structure, description, and even the HTTP call itself are the same.
Please try to find a way to make this more DRY, and easier to read. 

Hint: 
You could prepare some kind of list of tests to do for each property, and use it to "generate" test cases. Use some kind of array and a  `map()` method